releasenotes/notes/deprecate-extensions-d7ab3fca19962e2e.yaml

```diff
@@ -0,0 +1,40 @@
+---
+deprecations:
+  - |
+    To streamline the structure of Qiskit's gates and operations, the :mod:`qiskit.extensions`
+    module is pending deprecation and will be deprecated in a future release.
+    The following objects have been moved to :mod:`qiskit.circuit.library`
```

Comment:
All of `qiskit.circuit.library` is accessible from `qiskit.extensions` as well via a star import - it might have been good to mention that _all_ not-immediately-deprecated gates from `qiskit.extensions` should be imported from `qiskit.circuit.library`.